flipperdevices/flipperzero-firmware
# [FL-3889] 5V on GPIO control for ext. modules

# What's new

- Add support for OTG (5V) on GPIO:
    - control and get status via RPC.
    - control via Expansion protocol.
- Update the `expansion_test` app to test the OTG feature.
- Fixes #3537

Requires the [proto files](https://github.com/flipperdevices/flipperzero-protobuf/pull/51)  to be merged first.

# Verification 

1. Flash the firmware and disconnect the debug interface, in case it was connected (it will interfere with the next steps).
2. Read the instructions for the [expansion_test](https://github.com/flipperdevices/flipperzero-firmware/blob/dev/applications/debug/expansion_test/expansion_test.c) application and set up the test device accordingly.
3. Connect an LED (with an appropriate series resistor) to pins 1 (5V) and 8 (GND).
4. Build the `expansion_test` app and upload it to Flipper by running `./fbt launch APPSRC=expansion_test`. The app should run normally (see what it means in the instructions).
5. Disconnect the USB cable from Flipper. The LED should turn off.
6. Navigate to `Apps -> Debug -> Expansion Module Test` and run the app again. The LED should blink 2 times with ON and OFF times = 1 second.
7. Ensure that there are no regressions in the operation of prior expansion modules (e.g. VGM).

# Checklist (For Reviewer)
